presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -171,7 +171,7 @@ public GlueHiveMetastore(
     {
         this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
         this.hdfsContext = new HdfsContext(ConnectorIdentity.ofUser(DEFAULT_METASTORE_USER));
-        this.glueClient = requireNonNull(createAsyncGlueClient(glueConfig, requestHandler), "glueClient is null");
+        this.glueClient = createAsyncGlueClient(requireNonNull(glueConfig, "glueConfig is null"), requestHandler);
```

Comment:
This null check isn't needed, as we'll provide an NPE directly to the caller.

If you feel strongly that we should have an explicit check, then move it to the top so that it's standalone, as that seems cleaner than picking the first usage (there are other usages of it here). Do it as the first commit in this PR (before adding new code).

Reply:
> If you feel strongly that we should have an explicit check, then move it to the top so that it's standalone

Move to bottom